enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,366 @@
+Additionally, the network CRD will be modified in order to expose a new transport field to determine if encapsulation
+should be used for east/west traffic.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+#### Standalone Clusters
+
+#### Single-node Deployments or MicroShift
+
+### Risks and Mitigations
+
+BGP has a wide range of configurations and configuration options that are supported by FRR today. There is a big risk
+of scope creep to try to support all of these options during the initial development phase. During the initial release
+the number of options supported will be limited to mitigate this issue, and the main focus will be on stability of a
+default environment without enabling these extra options/features.
+
+Reliance on FRR is another risk, with no presence from the OCP networking team involved in that project to add support
+for OVN/OVS.
+
+Another risk is integration with Metal LB. We need to ensure that Metal LB is still able to function correctly with
+whatever OVN-Kubernetes is configuring within FRR.
+
+In addition, by dynamically learning routes and programming them into OVN, we risk a customer accidentally introducing
+potentially hundreds or even thousands of routes into different OVN logical routers. As a mitigation, we can recommend
+using communities or route aggregation to customers to limit RIB size. We will also need to scale test the effect of
+many routes inside OVN logical routers.
+
+Also, FRR-K8S is not GA yet, and targeted for 4.17.
+
+### Drawbacks
+
+Increased complexity of our SDN networking solution to support more complex networking. Additionally, it increases
+support complexity, because we are peering with a user’s fabric now and no longer blindly tunneling through their
+network with Geneve.
+
+### Implementation Details/Notes/Constraints
+
+#### BGP Peering
+
+FRR-K8S shall support both internal BGP (iBGP) and external BGP (eBGP). With iBGP, a full mesh topology is required. This
+can be cumbersome and have issues with scaling in large clusters. Therefore, FRR-K8S shall also support designating one
+or more nodes as route reflectors. With route reflectors, nodes only have to connect to the route reflectors and do not
+have to maintain a full mesh with all other nodes in the cluster.
+
+#### FRR-K8S Integration
+
+As previously mentioned frr-k8s will be used in order to deploy and manage FRR configuration. Support will be added
+where necessary for gaps in frr-k8s API, but in the meantime frr-k8s supports also issuing raw FRR configuration. The
+raw configuration will be for things that OVN-Kubernetes needs to configure, and not something exposed to a user. A user
+will need the ability to:
+
+* Define BGP peers, ASN, along with other common BGP configurations like communities, route filtering, etc.
+* Configure BFD
+
+For standard BGP support with the default VRF, frr-k8s API can be used in order to configure everything. In this case a
```

Comment:
is this related to the learning routes section?

Also, FRR-K8s supports VRFs too.

Reply:
I think this was a relic from the combined EVPN/BGP doc. I'll reword or remove this.